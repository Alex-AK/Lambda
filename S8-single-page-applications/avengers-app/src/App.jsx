import React, { Component } from 'react';
import { Route, Link } from 'react-router-dom';
import './App.css';

import avengers from './data';

import AvengersList from './components/AvengersList';
import Home from './components/Home';
import AvengerPage from './components/AvengerPage';

class App extends Component {
  constructor() {
    super();
    this.state = {
      avengers: avengers
    };
  }

  // componentDidMount = () => {
  //   this.setState({
  //     avengers: avengers
  //   });
  // };

  render() {
    console.log(avengers);
    return (
      <div className="App">
        <ul className="navbar">
          <li>
            <Link to="/">Home</Link>
          </li>
          <li>
            <Link to="/avengers">Avengers</Link>
          </li>
        </ul>
        <Route exact path="/" component={Home} />
        <Route exact path="/avengers" component={AvengersList} />
        <Route path="/avengers/:id" component={AvengerPage} />
      </div>
    );
  }
}

export default App;
