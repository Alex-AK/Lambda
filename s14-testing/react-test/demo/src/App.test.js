import React from 'react';
import { render, fireEvent, cleanup } from 'react-testing-library';
import App from './App';
import renderer from 'react-test-renderer';

import 'react-testing-library/cleanup-after-each'; // cleans up after all tests across app
import 'jest-dom/extend-expect';

// scoped to test file
afterEach(cleanup);

describe('<App />', () => {
  it('renders without crashing', () => {
    render(<App />);
    // scoped to individual describe
    afterEach(cleanup);
  });

  it('renders "hello world"', () => {
    const { getByText } = render(<App />);

    const actual = getByText(/hello world/i);

    expect(actual).toBeInTheDocument();
  });

  it('greet the team', () => {
    const { getByText } = render(<App />);

    const greetButton = getByText(/click me/i);
    fireEvent.click(greetButton);

    expect(getByText(/hello!/i)).toBeInTheDocument();
  });

  it.skip('matches snapshot', () => {
    const tree = renderer.create(<App />);
    expect(tree.toJSON()).toMatchSnapshot();
  });

  describe('mocking', () => {
    it('is mocking', () => {
      const mock = jest.fn(() => 'hello'); // shortcut for .mockImplementation
      // const mock = jest.fn().mockImplementation(() => 'hello');
      // const mock = jest.fn().mockReturnValue('hello');

      const actual = mock('hello');

      // running assertions against the value, not the function
      // expect(actual).toBeUndefined();
      expect(actual).toBe('hello');
      expect(mock).toHaveBeenCalled();
      expect(mock).toHaveBeenCalledTimes(1);
      expect(mock).toHaveBeenCalledWith('hello');
      expect(mock).not.toHaveBeenCalledWith('frown');
    });
    describe('speak()', () => {
      it('update the message when speak button is clicked', () => {
        const { getByText, queryByText } = render(<App />);

        const button = getByText(/speak/i);
        expect(queryByText(/this is a new message/i)).toBeFalsy();

        fireEvent.click(button);
        expect(queryByText(/this is a new message/i)).toBeTruthy();
      });
    });
  });
});
