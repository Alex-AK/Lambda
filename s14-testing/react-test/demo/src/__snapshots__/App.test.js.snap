// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> matches snapshot 1`] = `
<div
  className="App"
>
  <h2>
    Hello World
  </h2>
  <div>
    <button
      onClick={[Function]}
    >
      click me
    </button>
    <p>
      
    </p>
  </div>
</div>
`;
